### PR TITLE
Angular: Fix Angular sandbox startup errors

### PR DESCRIPTION
**Pull request by kroeder**
## What I did

In `main,` you cannot bootstrap a startable Angular Sandbox app. The Angular app type-checker complains that many `ts-expect-errors` are unnecessary and must be removed. Angular is wrong here, but I don't know how to tell Angular that these `@ts-expect-errors` were right. But I also think all the `ts-expect-errors` were unnecessary and solvable in non-invasive ways.

This PR aims to make the Angular Sandbox startable again without any workarounds.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

1. Create an Angular sandbox app
2. Start it
3. It should start without errors

**Sanity check:**
1. Use `main`
2. Create an Angular sandbox app
3. Start it
4. It should fail to start because of 5 `@ts-expect-errors`

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Comment by kroeder** [comment]
@valentinpalkovic are these the "hick ups" you mentioned when running a sandbox for Angular?

**Review by valentinpalkovic — APPROVED** [review]
LGTM! Thanks for finally making the Angular sandbox experience great again! :)

**Comment by kroeder** [comment]
@valentinpalkovic my last pr was ages ago.. Do I merge myself after approve or do core maintainers usually do it?

**Comment by valentinpalkovic** [comment]
Usually the core maintainers will merge the PRs :)

**Pull request merged**
